Sources/JWTKit/RSA/RSA.swift

```diff
@@ -0,0 +1,263 @@
+        ///     - pem: Contents of PEM file.
+        public init(pem data: some DataProtocol) throws {
+            let string = String(decoding: data, as: UTF8.self)
+            try self.init(pem: string)
+        }
+
+        /// Initializes a new ``RSA.PrivateKey`` instance with modulus, exponent, and private exponent.
+        ///
+        /// This convenience initializer creates an RSA key using the modulus, exponent, and private exponent.
+        /// All these parameters are expected to be base64 URL encoded strings.
+        ///
+        /// - Parameters:
+        ///   - modulus: The modulus of the RSA key, represented as a base64 URL encoded string.
+        ///   - exponent: The exponent of the RSA key, represented as a base64 URL encoded string.
+        ///
+        /// - Throws:
+        ///   - ``JWTError/generic`` with the identifier `RSAKey` if either the modulus or exponent cannot be decoded from their base64 URL encoded strings.
+        ///   - ``RSAError/keyInitializationFailure`` if there is a failure in initializing the RSA key, especially when the private key components are involved.
+        ///
+        /// - Note:
+        ///   - The provided modulus and exponent are key components for creating RSA public keys.
+        ///   - The private exponent is an additional parameter required for creating RSA private keys.
+        public init(
+            modulus: String,
+            exponent: String,
+            privateExponent: String
+        ) throws {
+            let n = try decode(modulus)
+            let e = try decode(exponent)
+            let d = try decode(privateExponent)
+
+            var serializer = DER.Serializer()
+
+            guard let privateKeyDER = try RSA.calculatePrivateDER(n: n, e: e, d: d) else {
+                throw RSAError.keyInitializationFailure
+            }
+            try privateKeyDER.serialize(into: &serializer)
+            let privateKey = try _RSA.Signing.PrivateKey(derRepresentation: serializer.serializedBytes)
+            self.init(backing: privateKey)
+        }
+
+        init(backing: _RSA.Signing.PrivateKey) {
+            self.backing = backing
+            self.publicKey = .init(backing: backing.publicKey)
+        }
+
+        func signature<D: Digest>(for digest: D, padding: _RSA.Signing.Padding) throws -> _RSA.Signing.RSASignature {
+            try self.backing.signature(for: digest, padding: padding)
+        }
+    }
+}
+
+extension RSA {
+    // Helper function to decode base64URL strings
+    private static func decode(_ string: String) throws -> Data {
+        guard let data = string.base64URLDecodedData() else {
+            throw JWTError.generic(identifier: "RSAKey", reason: "Unable to decode base64url string: \(string)")
+        }
+        return data
+    }
+}
+
+extension RSA.PrivateKey: Equatable {
+    public static func == (lhs: Self, rhs: Self) -> Bool {
+        lhs.backing.derRepresentation == rhs.backing.derRepresentation
```

Comment:
The `derRepresentation` of a key is likely to be expensive to generate, and an `Equatable` type is expected to have its equality operator invoked relatively often. Not to mention that depending on the backing implementation (which you should not assume to behave any particular way), there may be more than one valid DER encoding that is not precisely byte-equal (although that would normally be a BER or CER encoding, it does occasionally crop up). Neither CryptoKit nor SwiftCrypto's overlay makes `ECPrivateKey` or `ECPublicKey` equatable; do we really need this for the RSA keys?

Reply:
Nope, makes sense to put it into the tests only, that's where we're using it